lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -740,6 +797,88 @@ ParseResult FIRParser::parseOptionalRUW(RUWAttr &result) {
   return success();
 }
 
+ParseResult FIRParser::importAnnotations(const SMLoc &loc,
+                                         StringRef annotationsStr,
+                                         MLIRContext *context,
+                                         StringRef errorMsgParse) {
+
+  auto annotations = llvm::json::parse(annotationsStr);
+  if (auto err = annotations.takeError()) {
+    handleAllErrors(std::move(err), [&](const llvm::json::ParseError &a) {
+      auto diag = emitError(loc, errorMsgParse);
+      diag.attachNote() << a.message();
+    });
+    return failure();
+  }
+
+  llvm::json::Path::Root root;
+  llvm::StringMap<ArrayAttr> annotationMap;
+  if (annotations) {
```

Comment:
I'd hoist the "if annotations" check up a bit, invert it and make it an early out to simplify/unnest the logic:

`if (!annotations) return success();`

Reply:
Good thought. Looking at this again, this can actually just be removed. This is already being checked for errors above and blind `annotations.get()` will succeed, unless I misunderstand how `llvm::Expected` works...